tests/tests/runner.rs

```diff
@@ -33,81 +35,119 @@ use graph_core::{
 };
 use graph_mock::MockMetricsRegistry;
 use graph_node::manager::PanicSubscriptionManager;
-use graph_node::{
-    config::{Config, Opt},
-    store_builder::StoreBuilder,
-};
+use graph_node::{config::Config, store_builder::StoreBuilder};
+use graph_tests::helpers::run_cmd;
 use slog::{debug, info, Logger};
-use std::env;
+use std::process::Command;
 
 #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
-async fn test_runner() -> anyhow::Result<()> {
-    let subgraph_name = SubgraphName::new("test1")
+async fn data_source_revert() -> anyhow::Result<()> {
+    let subgraph_name = SubgraphName::new("data-source-revert")
         .expect("Subgraph name must contain only a-z, A-Z, 0-9, '-' and '_'");
 
-    let deployment = DeploymentLocator {
-        id: DeploymentId::new(1),
-        // ethereum-blocks
-        hash: DeploymentHash::new("QmUuqWUHTNaHkq6d333n7gCrZbwzLN2yophaHeeK9x6BjJ")
-            .expect("unable to parse hash"),
+    let ipfs = IpfsClient::localhost();
+    ipfs.test()
+        .await
+        .expect("Could not connect to IPFS, make sure it's running at port 5001");
+
+    // This setup can be run once for all tests.
+    run_cmd(Command::new("yarn").current_dir("./integration-tests"));
+
+    let test_dir = format!("./integration-tests/{}", subgraph_name);
+    run_cmd(Command::new("yarn").arg("codegen").current_dir(&test_dir));
+
+    let deploy_output = run_cmd(
+        Command::new("yarn")
+            .arg("deploy:test")
+            .env("IPFS_URI", "http://127.0.0.1:5001")
+            .env("GRAPH_NODE_ADMIN_URI", "http://localhost:4242")
+            .current_dir(test_dir),
+    );
+
+    // Hack to extract deployment id from `graph deploy` output.
+    const ID_PREFIX: &str = "Build completed: ";
+    let hash = {
+        let mut line = deploy_output
+            .lines()
+            .find(|line| line.contains(ID_PREFIX))
+            .expect("found no matching line");
+        if !line.starts_with(ID_PREFIX) {
+            line = &line[5..line.len() - 5]; // workaround for colored output
+        }
+        DeploymentHash::new(line.trim_start_matches(ID_PREFIX)).unwrap()
+    };
```

Comment:
Prolly worth moving most stuff up to this line into a common::setup to be re-usable but also help read the test content

Reply:
Added a `fixture` module and moved that there.